src/tests/sqlsmith/src/expr.rs

```diff
@@ -31,35 +34,63 @@ lazy_static::lazy_static! {
     };
 }
 
+lazy_static::lazy_static! {
+    static ref AGG_FUNC_TABLE: HashMap<DataTypeName, Vec<AggFuncSign>> = {
+        init_agg_table()
+    };
+}
+
 fn init_op_table() -> HashMap<DataTypeName, Vec<FuncSign>> {
     let mut funcs = HashMap::<DataTypeName, Vec<FuncSign>>::new();
     func_sigs().for_each(|func| funcs.entry(func.ret_type).or_default().push(func.clone()));
     funcs
 }
 
+fn init_agg_table() -> HashMap<DataTypeName, Vec<AggFuncSign>> {
+    let mut funcs = HashMap::<DataTypeName, Vec<AggFuncSign>>::new();
+    agg_func_sigs().for_each(|func| funcs.entry(func.ret_type).or_default().push(func.clone()));
+    funcs
+}
+
 impl<'a, R: Rng> SqlGenerator<'a, R> {
-    pub(crate) fn gen_expr(&mut self, typ: DataTypeName) -> Expr {
+    pub(crate) fn gen_expr(&mut self, typ: DataTypeName, can_agg: bool, inside_agg: bool) -> Expr {
```

Comment:
Add docs about `can_agg` and `inside_agg`. It isn't immediately apparent what they are, what constraints they have on generated expressions.

Maybe something like:
- `gen_expr` has two execution modes. One where it generates aggregates, another where it doesn't. This is determined by `can_agg` flag. These execution modes are required because ...
- When generating aggregate expressions, there are also additional constraints on generated expressions within them. Hence `inside_agg` is used to provide this context.